src/components/Grid/Pagination.js

```diff
@@ -0,0 +1,130 @@
+import React, { useEffect, useState } from 'react'
+import PropTypes from 'prop-types'
+import styles from './Pagination.module.css'
+
+const usePagination = ({
+  data = null,
+  total = null,
+  per_page = null,
+  total_pages = null,
+  current_page = 1,
+  fetchPage = null,
+  startPageNumber = 1,
+}) => {
+  if (!fetchPage) {
+    throw Error('usePagination requires a fetchPage callback')
+  }
+
+  const [pagingState, setPagingState] = useState({
+    data,
+    total,
+    per_page,
+    total_pages,
+    current_page,
+    fetchPage,
+  })
+
+  const fetchPageDelegate = async (pageNumber) => {
+    const data = await fetchPage(pageNumber)
+    setPagingState({
+      data: data.data,
+      total: data.total,
+      per_page: data.per_page,
+      current_page: data.page,
+      total_pages: data.total_pages,
+    })
+  }
+
+  const getPaginationNumbers = () => {
+    const pageNumbers = []
+    let paginationNumbers
+    if (pagingState.total !== null) {
+      for (
+        let i = 1;
+        i <= Math.ceil(pagingState.total / pagingState.per_page);
```

Comment:
why is this logic needed when we have a variable `total_pages` ? 

Also, there's no check `per_page != null`.

Reply:
Ha, yes, you're correct! I think I had it in before I elected to add the `total_pages` and also hadn't got buy in for that, but I'll switch to using `total_page` here 👍 

For the null check 1. I'll be switching to `total_pages` so it'll be a bit of a moot point, but, also, I think the server must always return `per_page` so the check for `pagingState.total` on line 41 makes it safe to assume an http response happened, and since the contract is to always supply that I don't know how rigorous we need to check for everything 🤔